src/main/java/io/confluent/connect/jdbc/source/TimestampIncrementingTableQuerier.java

```diff
@@ -146,7 +146,17 @@ protected void createPreparedStatement(Connection db) throws SQLException {
     stmt = db.prepareStatement(queryString);
   }
 
-  @Override
+  private void appendWhereIfNotPresent(StringBuilder builder) {
+	  if(query != null && !query.trim().isEmpty()){
+		  if(query.indexOf(" where ") != -1){
+			  builder.append(" AND ");
+		  }else{
+			  builder.append(" WHERE ");
+		  }
+	  }
```

Comment:
This no longer appends ` WHERE ` for the `TABLE` case (e.g., when `query == null || query.trim().isEmpty()`).

Reply:
Perhaps a more straightforward, more efficient, and less error prone approach is to use the switch statement starting on line 87 to determine whether these criteria should be prefixed with a `WHERE` or an `AND`. Something like:

    String criteriaStart = " WHERE ";
    switch (mode) {
       case TABLE:
         builder.append("SELECT * FROM ");
         builder.append(JdbcUtils.quoteString(name, quoteString));		         
         break;
       case QUERY:
         builder.append(query);
         if (query.toUpperCase().contains(" WHERE ")) {
             criteriaStart = " AND ";
         }
         break;
       default:
         throw new ConnectException("Unknown mode encountered when preparing query: " + mode.toString());
     }
 		
and then replace the:

    builder.append(" WHERE ");

lines with

    builder.append(CRITERIA_START);